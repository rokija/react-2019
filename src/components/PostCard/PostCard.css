.PostCard__card-title {
    font-weight: 700;
}
