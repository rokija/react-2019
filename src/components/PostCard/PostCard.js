import React, { Component } from "react";
import { Card, CardImg, CardText, CardBody, CardTitle } from 'reactstrap';
import "./PostCard.css";

class PostCard extends Component {
    render() {
        const { title, description, image } = this.props;

        return (
            <Card className="PostCard mb-5">
                <CardImg top width="100%" src={image} alt="Card image cap" />
                <CardBody>
                    <CardTitle className="PostCard__card-title">{title}</CardTitle>
                    <CardText>{description}</CardText>
                </CardBody>
            </Card>
        )
    }
}

export default PostCard;
