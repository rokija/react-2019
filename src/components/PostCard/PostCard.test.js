import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import PostCard from "./PostCard";

describe("PostCard component", () => {
  const component = shallow(<PostCard />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
