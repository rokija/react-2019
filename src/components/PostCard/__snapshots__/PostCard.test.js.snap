// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard component renders properly 1`] = `
<Card
  className="PostCard mb-5"
  tag="div"
>
  <CardImg
    alt="Card image cap"
    tag="img"
    top={true}
    width="100%"
  />
  <CardBody
    tag="div"
  >
    <CardTitle
      className="PostCard__card-title"
      tag="div"
    />
    <CardText
      tag="p"
    />
  </CardBody>
</Card>
`;
